main.go

```diff
@@ -176,6 +176,34 @@ func init() {
 	}
 }
 
+//Defines the interface for removing orphaned artifacts from aborted scans
+type CleanTemp interface {
+	//Removes orphaned directories from sources like Git
+	CleanTempDir(ctx context.Context, dirName string, pid int) error
+	//Removes orphaned files/artifacts from sources like Artifactory
+	CleanTempFiles(ctx context.Context, fileName string, pid int) error
+}
+
+func CleanTempDir(ctx context.Context, dirName string, pid int) error {
+	tempDir := os.TempDir()
+	files, err := os.ReadDir(tempDir)
+	if err != nil {
+		return fmt.Errorf("Error reading temp dir: %w", err)
+	}
+
+	pidStr := strconv.Itoa(pid)
+
+	for _, file := range files {
+		if file.IsDir() && strings.Contains(file.Name(), dirName) && !strings.Contains(file.Name(), pidStr) {			dirPath := fmt.Sprintf("%s/%s", tempDir, file.Name())
```

Comment:
**Nit:** [filepath.Join](https://pkg.go.dev/path/filepath#Join) can be used to handle OS-specific separators.